prov/rxm/src/rxm_ep.c

```diff
@@ -1260,7 +1262,7 @@ rxm_ep_format_tx_inject_iov(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn,
 			    uint64_t data, uint64_t flags, uint64_t tag,
 			    uint8_t op, uint64_t comp_flags, struct rxm_tx_buf **tx_buf)
 {
-	if (OFI_UNLIKELY(rxm_ep_format_tx_res_lightweight(
+	if (OFI_UNLIKELY(rxm_ep_format_tx_buf(
 				rxm_ep, rxm_conn, len, data, flags, tag, tx_buf,
 				&rxm_ep->buf_pools[RXM_BUF_POOL_TX_INJECT]))) {
 		return -FI_EAGAIN;
```

Comment:
Would defining a new rxm_tx_sar_buf instead of having a separate footer work?

Reply:
Similar comment for eager and rndv buffers as well.

Reply:
Done